### ISSUE TITLE
Could you please provide the training logs?

### ISSUE THREAD

**Issue by felixfuu** [microsoft/FIBER#9]
I try to reproduce the **coarse_grained** part, there are two problems:
1. When set per_gpu_batchsize=8, it is very easy to OOM on V100, even if it is set to 4;
2. After running 1000 iterations, the loss is still nan, and the model seems unable to converge. (The training command is: python run.py with nmetrium_gpus=7 num_nodes=1 task_pretrain_mlm_itm_itc per_gpu_batchsize=2)
`Epoch 0:   0%|                    | 1299/3995321 [15:26<791:22:06,  1.40it/s, loss=nan, v_num=1`

**Comment by felixfuu** [comment]
On the A100( 40GB), using the per_gpu_batchsize=8 is still OOM.

**Comment by zdou0830** [comment]
Hi, we used A100-80G when setting per_gpu_batchsize=0, but having a smaller per_gpu_batchsize is fine as the code will do gradient accumulation automatically.  

The loss will be nan initially but should be fine after a while. If not, you can check if these lines are causing the problems and fix them accordingly (e.g.  x = x/(x.norm(dim=-1, keepdim=True)+1e-9)
https://github.com/microsoft/FIBER/blob/main/coarse_grained/fiber/modules/fiber_module.py#L264 https://github.com/microsoft/FIBER/blob/main/coarse_grained/fiber/modules/fiber_module.py#L296).

We didn't keep the log and I don't have the resources to train it now.

**Comment by felixfuu** [comment]
@zdou0830 After training for 16 hours with 24-V100GPU , the loss is still nan.
Epoch 0:   6%|▋         | 75478/1165302 [16:37:52<240:08:20,  1.26it/s, loss=nan, v_num=0]

**Comment by zdou0830** [comment]
could you check if the abovementioned lines are causing the problem? also, ideally you can use a power of 2 gpus (e.g. 16) because the batch size is 4096.

**Comment by zdou0830** [comment]
could you replace those lines with x = x/(x.norm(dim=-1, keepdim=True)+1e-9) and see if the problem exists?

**Comment by felixfuu** [comment]
> could you replace those lines with x = x/(x.norm(dim=-1, keepdim=True)+1e-9) and see if the problem exists?

Yes, I replaced those lines and the problem persists.

**Comment by zdou0830** [comment]
Thanks! It's a bit weird that mlm_labels will all be -100. are those empty sentences?

**Comment by felixfuu** [comment]
@zdou0830 I read your code in detail. It is because when the sentence is very short(and batch size is very small), the **mlm_collator** function will make each token mask out with a probability of 0.15, so it is possible that all tokens will be masked.

**Comment by felixfuu** [comment]
@zdou0830 I re-wrote a CustomDataCollatorForLanguageModeling to solve this problem, thanks for your reply~

**Comment by felixfuu** [comment]
@zdou0830 https://penzhanwu2v2.blob.core.windows.net/data/htzhang/coco/annotations/lvis_od_train.json This file has expired and cannot be downloaded, could you re-upload it?
